crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/meta_net.rs

```diff
@@ -57,6 +58,11 @@ kitsune_p2p_types::write_codec_enum! {
     }
 }
 
+fn next_msg_id() -> u64 {
+    static MSG_ID: std::sync::atomic::AtomicU64 = std::sync::atomic::AtomicU64::new(1);
```

Comment:
i'm curious to understand what the effects are if the conductor restarts quickly. will this number resetting be confusing to other peers? or is it purely local? i see it's being stored in `res_store` but i haven't looked up what that does.

Reply:
```suggestion
    // MAYBE - track these message ids at the connection level to prevent mismatches
    static MSG_ID: std::sync::atomic::AtomicU64 = std::sync::atomic::AtomicU64::new(1);
```

Reply:
Yeah, that's a good point. But to solve it I'll probably need to adjust the protocol slightly under the hood. And I've got a different issue in the multiplexing logic that might make sense to solve at the same time, so I'll defer this to follow-on work.

Reply:
thanks. i'll leave it to you to commit your suggestion and resolve this thread

Reply:
done